cmake/branding/recolor_graphics.sh.in

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+echo "Detecting build type..."
+readarray -d - -t TAGS_LIST <<< "@FORCE_VERSION@" # Splits FORCE_VERSION into an array called TAGS_LIST with - as the delimiter
```

Comment:
This is where I think we can just set the build type in cmake instead so that all of the parsing logic is contained inside that `VersionInfo.cmake` file.

Also, `BUILD_TYPE` is sort of reserved for how we handle Debug/Release build flags.  Perhaps we rename this to `RELEASE_TYPE`?

Next, one advantage to keeping this in CMake is it can be passed to the build system so that it eventually shows on the About screen, etc.

Finally, I think we should fallback to "NIGHTLY", not "UNKNOWN", don't you?

Reply:
> This is where I think we can just set the build type in cmake instead so that all of the parsing logic is contained inside that VersionInfo.cmake file.

I can move the type detection to `Versioninfo.cmake`, but that will involve adding a variable so it doesn't need to be parsed again elsewhere to function properly.

> Also, BUILD_TYPE is sort of reserved for how we handle Debug/Release build flags. Perhaps we rename this to RELEASE_TYPE?

Yikes, I better change that!

> Next, one advantage to keeping this in CMake is it can be passed to the build system so that it eventually shows on the About screen, etc.

Yes. Not seeing the input is making testing frustrating.

> Finally, I think we should fallback to "NIGHTLY", not "UNKNOWN", don't you?

No. "UNKNOWN" is the red "danger" fallback color. If the build is meant to be released, it will never reach this fallback.

Reply:
> No. "UNKNOWN" is the red "danger" fallback color. If the build is meant to be released, it will never reach this fallback.

I don't understand this at all.  It's either stable or nightly currently.  Perhaps we add a color for nightlies.  There's no danger icon, I'm not sure what this is aiming to achieve and I thought this was already discussed in #3347.  Anyway, it's easy enough to remove, but it needs to be decided.

>  but that will involve adding a variable

👍 This is a great use-case for a new variable because then we can eventually export it to a C++ header file (using a technique called `cmake define` for the About dialog.

Reply:
>I don't understand this at all. It's either stable or nightly currently. Perhaps we add a color for ~~nightlies~~ RC's. There's no danger icon, I'm not sure what this is aiming to achieve and I thought this was already discussed in #3347. Anyway, it's easy enough to remove, but it needs to be decided.

We did already discuss this, but perhaps we didn't end on the same page. Here's my thinking:

💚 `"STABLE"`: Meant for **general audiences**. Has no stage.
💙 `"BETA"`: Meant for **release candidates**. Has a "-rc" stage.
💛 `"NIGHTLY"` Meant for **nightly builds**. Will be designated by some sort of stage (assumed to be "-n", but that can change).
❤️ `"UNKNOWN"`: A build type was unable to be parsed from the provided info, meaning it was **not meant for public release**. This is the fallback, since the other three *are* meant for public release.

Reply:
So the only factor in making it red is that there was a parsing bug?  So LMMS releases that don't have a version at all?  What scenario?  Perhaps you're looking to safeguard against non-sanctioned builds, but this isn't the way to do that (signing our installers is, e.g. #3601 #3602 (e.g. something cryptographically verified).

Reply:
Either a parsing bug, or a non-tagged build. It's not a safeguard against non-sanctioned builds, just a reminder that the build can potentially be more unstable than a nightly. Regardless of the source, a red build is one that isn't ready for general use.

Reply:
But what I see being described is a use-case for another color, not how we can reasonably identify it in code.  Opinionatedly I wholeheartedly disagree with 4 colors. 
* how many warning signs do we need?
* If it's yellow, check the About screen.  If it's not, it's stable.
* the yellow warning system is simple, easy to support and easy for users to process.
* it works with our existing (albeit a bit crappy) parsing system without any changes and it moves the PR forward.
* adding a 4th color is moot if we don't really know what triggers it.

So objectively, I really only disagree with the 4th color being chosen with no evidence, testing or even a valid description of when it will be triggered.  Perhaps our `VersionInfo.cmake` needs to be improved to know the difference between an upstream build and a build triggered from a fork?  I'd love that because right now the parsing is garbage for forked builds.  If the colors are so important, can we at least agree to fix the versioning first?

Reply:
> the red will remain "unknown"

Just an idea, but if we fix the version parsing to actually fall into this block, perhaps we can name it by the fork it was built from, so that the information can be provided in the UI.

Reply:
> how many warning signs do we need?

None. This entire pull request is 100% cosmetic.

> If it's yellow, check the About screen. If it's not, it's stable.

This is an option, though personally I would prefer the yellow indicate that while it is unstable, it is stable enough to earn the yellow (over red). That's just personal preference though.

> it works with our existing (albeit a bit crappy) parsing system without any changes and it moves the PR forward.

The PR is broken in many places right now but none of that stems from having a fourth color. I can remove the red, but the PR isn't going to move forward until I can educate myself on CMake more and fix the abundance of errors.

> adding a 4th color is moot if we don't really know what triggers it.

While the red is a fallback, it's mostly meant to designate test builds and the like. It's more than a mystery fallback.

> Just an idea, but if we fix the version parsing to actually fall into this block, perhaps we can name it by the fork it was built from, so that the information can be provided in the UI.

I love this idea, actually. That would solve any appearance of red being a mystery fallback color and provide it with much more meaning than "might crash, who knows?".